src/d3a/models/market/__init__.py

```diff
@@ -63,7 +71,7 @@ def __init__(self, time_slot=None, area=None, notification_listener=None, readon
         self.accumulated_trade_energy = 0
         if notification_listener:
             self.notification_listeners.append(notification_listener)
-
+        self.current_tick = 0
```

Comment:
What you could do, is instead of increasing the current_tick in parallel to the area, increase it only in the area and just assign it in the markets. That way there will be only one place where the current_tick is increased, and we are going to be saved by any potential errors that the current_tick of the market is out-of-sync with the current_tick of the area.

Reply:
Ok, this means though, that we again have a connection between area and market. Or do you mean that i parse it in the constructor? But the we have to increase it in the market, too.

Reply:
I meant to pass it as an argument of the update_clock method. In that case, this will be a simple setter (only sets the current_tick of the market to the current tick of the area)

Reply:
ahh, yes. makes sense. Cool, will implement it.